src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -545,11 +544,12 @@ private boolean hasEmptyLine(int startLine, int endLine) {
      */
     private boolean hasEmptyLineBefore(DetailAST token) {
         final int lineNo = token.getLineNo();
+        boolean result = false;
         if (lineNo == 1) {
-            return false;
+            return result;
         }
-        //  [lineNo - 2] is the number of the previous line because the numbering starts from zero.
-        final String lineBefore = getLines()[lineNo - 2];
+        //  [lineNo - 1] is the number of the previous line because the numbering starts from zero.
+        final String lineBefore = getLines()[lineNo - 1];
```

Comment:
This is wrong, token.getLineNo() returns lines in range <1, N>, getLines operates in lines in range <0, N-1>. So the line returned by token.getLineNo() is in getLines(token.getLineNo() - 1), previous line in getLines(token.getLineNo() - 2). Did you read the comment with understanding?